### PR TITLE
feat: Duplicate Structure

### PR DESCRIPTION
**Pull request by seebergs**
## Changes

Initial version

## Test

1. Get the ID of the source entity you want to duplicate, and an ID of a destination entity.
2. Run `python duplicate_structure.py <source_entity_id> <destination_entity_id>`
3. Verify that the structure was duplicated.

### CODE REVIEW

File: python/tools/duplicate_structure/duplicate_structure.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+session = ftrack_api.Session(auto_populate=False)
+
+
+class DuplicateStructure(object):
+    def __init__(self):
+        self.logger = logging.getLogger('com.ftrack.recipes.tools.duplicate_structure')
+        self.logger.setLevel(logging.INFO)
+
+        self.parser = self.args = None
```
Comment (hdd) [inline python/tools/duplicate_structure/duplicate_structure.py]: I would split the argument parsing from the actual code .
as in (pseudocode): 
```

args = parsearguments()
class = MyClass()
class.process(arg1, arg2)

```
as it makes it easier to read.

File: python/tools/duplicate_structure/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Duplicate Structure
+
+A command line tool that duplicates a structure of objects in ftrack.
+
+## Usage
+
+```cmd
+python duplicate_structure.py <source_entity_id> <destination_entity_id>
```
Comment (hdd) [inline python/tools/duplicate_structure/README.md:8]: **destination_entity_id** is the parent where the **source_entity_id** will be created under ?

File: python/tools/duplicate_structure/duplicate_structure.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+import argparse
+import itertools
+import time
+
+import ftrack_api
+import ftrack_api.attribute
+
+logging.basicConfig()
+
+session = ftrack_api.Session(auto_populate=False)
```
Comment (hdd) [inline python/tools/duplicate_structure/duplicate_structure.py:20]: add a comment on why initialising the session with **auto_populate** disabled

**Review by hdd — CHANGES_REQUESTED** [review]

**Comment by hdd** [comment]
@seebergs please add requirements.txt with required modules (ftrack-python-api only , likely)

File: python/tools/duplicate_structure/duplicate_structure.py (hunk 1 above)
Comment (seebergs) [reply python/tools/duplicate_structure/duplicate_structure.py]: Sure, copied this approach from the manage project schemas script but happy to change to this instead.

File: python/tools/duplicate_structure/README.md (hunk 2 above)
Comment (seebergs) [reply python/tools/duplicate_structure/README.md:8]: Exactly.

File: python/tools/duplicate_structure/duplicate_structure.py (hunk 3 above)
Comment (seebergs) [reply python/tools/duplicate_structure/duplicate_structure.py:20]: Added.

**Review by hdd — DISMISSED** [review]
great work here !

**Review by hdd — APPROVED** [review]

**Pull request merged**